derive_builder_macro/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+//! Derive a builder for a struct
+
+#![crate_type = "proc-macro"]
+#![deny(warnings)]
+
+#[macro_use]
+extern crate darling;
+
+extern crate proc_macro;
+extern crate proc_macro2;
+#[macro_use]
+extern crate syn;
+#[macro_use]
+extern crate quote;
+#[cfg(feature = "logging")]
+#[macro_use]
+extern crate log;
+extern crate derive_builder_core;
+#[cfg(feature = "logging")]
+extern crate env_logger;
+
+#[cfg(not(feature = "logging"))]
+#[macro_use]
+mod log_disabled;
+mod options;
+
+use darling::FromDeriveInput;
+use options::Options;
+use proc_macro::TokenStream;
+#[cfg(feature = "logging")]
+use std::sync::Once;
+
+#[cfg(feature = "logging")]
+static INIT_LOGGER: Once = Once::new();
+
+#[doc(hidden)]
+#[proc_macro_derive(Builder, attributes(builder))]
+pub fn derive(input: TokenStream) -> TokenStream {
+    #[cfg(feature = "logging")]
+    INIT_LOGGER.call_once(|| {
+        env_logger::init();
+    });
+
+    let ast = parse_macro_input!(input as syn::DeriveInput);
+
+    builder_for_struct(ast).into()
+}
+
+fn builder_for_struct(ast: syn::DeriveInput) -> proc_macro2::TokenStream {
```

Comment:
So this function should be moving to `derive_builder_core` and be made public; that way we can greatly restrict the surface area of the crate that exposes the main functionality. The macro crate then just calls that crate.